apps/hash-graph/docker/Dockerfile

```diff
@@ -20,14 +20,20 @@ RUN apk add --no-cache make openssl-dev protobuf-dev
 RUN --mount=type=cache,target=/usr/local/cargo/registry \
     --mount=type=cache,target=/usr/local/cargo/git \
     --mount=type=cache,target=/usr/local/src/hash-graph/target \
-    cargo install --path bin/cli --profile $PROFILE --locked
+    cargo install --path bin/cli --profile $PROFILE --features type-fetcher --locked
+
+RUN --mount=type=cache,target=/usr/local/cargo/registry \
+    --mount=type=cache,target=/usr/local/cargo/git \
+    --mount=type=cache,target=/usr/local/src/hash-graph/target \
+    cargo install --path bin/type-fetcher --profile $PROFILE --locked
 
 # Replace with `FROM scratch` when a health check command was added to hash-graph
 FROM alpine:3.17 AS runtime
 
 RUN addgroup -S appgroup && adduser -S appuser -G appgroup
 
 COPY --from=builder --chown=appgroup:appuser /usr/local/cargo/bin/hash-graph /
+COPY --from=builder --chown=appgroup:appuser /usr/local/cargo/bin/hash-type-fetcher /
```

Comment:
Not sure what I feel about having this as a shared dockerfile, I think I like it? It's just slightly unconventional I guess. Did you have any comments on this multi-binary dockerfile setup?

Reply:
The Dockerfile now only contains one binary (`FROM scratch`!) since https://github.com/hashintel/hash/pull/2038